File: android/src/main/java/com/hemangkumar/capacitorgooglemaps/MapPreferencesZoom.java
package com.hemangkumar.capacitorgooglemaps;

import androidx.annotation.Nullable;

import com.getcapacitor.JSObject;


public class MapPreferencesZoom  {

    public Integer minZoom;
    public Integer maxZoom;

    public static final String MIN_ZOOM_KEY = "minZoom";
    public static final String MAX_ZOOM_KEY = "maxZoom";

    public MapPreferencesZoom() {
        this.minZoom = 8;
        this.maxZoom = 22;
    }

    public void updateFromJSObject(@Nullable JSObject jsObject) {
        if (jsObject != null) {
            Integer minZoom = jsObject.getInteger(MIN_ZOOM_KEY, 8);
            if (minZoom != null) {
                this.minZoom = minZoom;
            }

            Integer maxZoom = jsObject.getInteger(MAX_ZOOM_KEY, 22);
            if (maxZoom != null) {
                this.maxZoom = maxZoom;
            }
        }
    }
}

File: ios/Plugin/MapPreferencesZoom.swift
import Capacitor
import GoogleMaps

class MapPreferencesZoom {
    public static let MIN_ZOOM_KEY: String! = "minZoom"
    public static let MAX_ZOOM_KEY: String! = "maxZoom"
    
    var minZoom = 8
    var maxZoom = 22
    
    func updateFromJSObject(object: JSObject) {
        if object[MapPreferencesZoom.MIN_ZOOM_KEY] != nil {
            if let minZoomUpdate = object[MapPreferencesZoom.MIN_ZOOM_KEY] as? Int {
                minZoom = minZoomUpdate
            }
        }
        
        if object[MapPreferencesZoom.MAX_ZOOM_KEY] != nil {
            if let maxZoomUpdate = object[MapPreferencesZoom.MAX_ZOOM_KEY] as? Int {
                maxZoom = maxZoomUpdate
            }
        }
    }
    
    func getJSObject(_ mapView: GMSMapView) -> JSObject {
        return [MapPreferencesZoom.MIN_ZOOM_KEY: mapView.minZoom, MapPreferencesZoom.MAX_ZOOM_KEY: mapView.maxZoom]
    }
}
